helix-core/src/numbers.rs

```diff
@@ -0,0 +1,144 @@
+use std::borrow::Cow;
+
+use ropey::{Rope, RopeSlice};
+
+pub struct NumberInfo {
+    pub start: usize,
+    pub end: usize,
+    pub value: i64,
+    pub radix: u32,
+}
```

Comment:
It would be good if it can also parse in the separator and separator digits. Say,
```
1_000
20,000
0b1000_1010
```

Reply:
That would be nice. In Vim it treats each part as separate numbers, but I think this would be better.

Reply:
This could get complicated with irregular separator spacing. For example, `1_23_456_7890` is a valid integer literal in Rust. It shouldn't be hard to detect that as a number, but in your example with `999_999` it adds a separator when it increments to `1_000_000`. But how should it increment something like `99_999_9`? Maybe it should only add separators if there is regular spacing.  If there isn't a regular spacing it could keep existing separators, but not add new ones. If that's the way we do it, `99_999_9` would increment to `100_000_0`.

Reply:
I've added support for separators.